### PR TITLE
Upgrade node to 18, Gatsby to 4

### PR DESCRIPTION
**Pull request by treycucco**
In this PR I upgraded node 12 -> 18 and Gatsby 2 -> 4.

I tried upgrading to Gatsby 5 but there were a lot of changes and issues and I didn't want to spend too much more time on this.

### CODE REVIEW

File: src/components/banner.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { graphql, Link } from 'gatsby';
 import PropTypes from 'prop-types';
-import styles from '../styles/banner.module.scss';
+import * as styles from '../styles/banner.module.scss';
```
Comment (treycucco) [inline src/components/banner.js:4]: css modules are imported as es modules.

https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#css-modules-are-imported-as-es-modules

File: src/pages/jobs.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ const Jobs = ({ data }) => {
           </div>
           <ul className={jobsStyles.jobslist}>
             {allLever.nodes.map((job) => (
-              <Job className={jobsStyles.jobs} key={job.id} job={job} />
```
Comment (treycucco) [inline src/pages/jobs.js:28]: There's no `jobs` class and it complains about that now that it's using es modules

File: src/templates/blogTemplate.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ const BlogTemplate = ({ data }) => {
         )}
       </Helmet>
 
-      <div className={styles.container}>
```
Comment (treycucco) [inline src/templates/blogTemplate.js:29]: No container class in css

File: .eslintrc.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "prettier/react",
     "plugin:react/recommended"
   ],
-  "parser": "babel-eslint",
```
Comment (treycucco) [inline .eslintrc.json:8]: babel-eslint is no longer supported and was giving various dependency upgrade complaints.

File: package.json [hunk 5]
```diff
@@ -60,7 +60,7 @@
     "eslint-plugin-standard": "^4.0.1",
     "gatsby-source-lever": "^4.25.0",
     "node-fetch": "^2.6.1",
-    "node-sass": "^4.14.1",
+    "node-sass": "^8.0.0",
```
Comment (treycucco) [inline package.json:63]: node-sass has very specific versions that work with specific node versions:

https://github.com/sass/node-sass#node-version-support-policy

**Review by lukasmlady — APPROVED** [review]
LGTM

**Pull request merged**